cli/exp_task_logs.go

```diff
@@ -71,8 +89,167 @@ func (r *RootCmd) taskLogs() *serpent.Command {
 			_, _ = fmt.Fprintln(inv.Stdout, out)
 			return nil
 		},
+		Options: serpent.OptionSet{
+			{
+				Flag:          "follow",
+				FlagShorthand: "f",
+				Description:   "Follow logs in real-time, similar to 'tail -f'.",
+				Value:         serpent.BoolOf(&follow),
+			},
+		},
 	}
 
 	formatter.AttachOptions(&cmd.Options)
 	return cmd
 }
+
+// TODO(cian): this will need to be updated when the task data model is updated.
+func (*RootCmd) followTaskLogs(inv *serpent.Invocation, client *codersdk.Client, taskID uuid.UUID) error {
+	ctx := inv.Context()
+	workspace, err := client.Workspace(ctx, taskID)
+	if err != nil {
+		return xerrors.Errorf("fetch workspace: %w", err)
+	}
+	if workspace.LatestBuild.ID == uuid.Nil {
+		return xerrors.Errorf("workspace has no builds")
+	}
+	build := workspace.LatestBuild
+	if build.HasAITask == nil || !*build.HasAITask {
+		return xerrors.Errorf("workspace is not configured as an AI task")
+	}
+
+	if build.AITaskSidebarAppID == nil || *build.AITaskSidebarAppID == uuid.Nil {
+		return xerrors.Errorf("task is not configured with a sidebar app")
+	}
+
+	sidebarAppID := *build.AITaskSidebarAppID
+	var agentID uuid.UUID
+	var sidebarApp *codersdk.WorkspaceApp
+
+	for _, res := range build.Resources {
+		for _, agent := range res.Agents {
+			for _, app := range agent.Apps {
+				if app.ID == sidebarAppID {
+					agentID = agent.ID
+					sidebarApp = &app
+					break
```

Comment:
Would breaking to a label be blasphemous (instead of the `if`s below)?  I know gotos get a bad rap though 😆